### PR TITLE
[REF] stock: _gather use now the ORM

### PR DESCRIPTION
**Pull request by ryv-odoo**
Remove the feature of `NULL` order of stock.quant and put
the `in_date` required because it is always set (and should).

odoo/upgrade#2019
task-2413099

### CODE REVIEW

**Comment by Whenrow** [comment]
Maybe rewrite the test_in_date_4b (at least comments)

**Comment by amoyaux** [comment]
robodoo r+

File: addons/stock/models/stock_quant.py [hunk 1]
```diff
@@ -92,7 +92,7 @@ def _domain_product_id(self):
         'Available Quantity',
         help="On hand quantity which hasn't been reserved on a transfer, in the default unit of measure of the product",
         compute='_compute_available_quantity')
-    in_date = fields.Datetime('Incoming Date', readonly=True)
+    in_date = fields.Datetime('Incoming Date', readonly=True, required=True, default=fields.Datetime.now())
```
Comment (Feyensv) [inline addons/stock/models/stock_quant.py:95]: The default value will always be the same, i.e. `Datetime.now` value when launching the server.
It should be either `default=lambda self: fields.Datetime.now()` or `default=fields.Datetime.now` ;)

Comment (ryv-odoo) [reply addons/stock/models/stock_quant.py:95]: Oups, my bad. You alright! Thx

Comment (ryv-odoo) [reply addons/stock/models/stock_quant.py:95]: I am going to create a PR fix!

Comment (ryv-odoo) [reply addons/stock/models/stock_quant.py:95]: odoo/odoo#63387

**Pull request closed without merging**